types/wlr_drm.c

```diff
@@ -36,9 +102,39 @@ static void drm_handle_create_prime_buffer(struct wl_client *client,
 		struct wl_resource *resource, uint32_t id, int fd, int32_t width,
 		int32_t height, uint32_t format, int32_t offset0, int32_t stride0,
 		int32_t offset1, int32_t stride1, int32_t offset2, int32_t stride2) {
-	close(fd);
-	wl_resource_post_error(resource, WL_DRM_ERROR_INVALID_NAME,
-		"DMA-BUF imports via wl_drm are not supported, use linux-dmabuf instead");
+	struct wlr_dmabuf_attributes dmabuf = {
+		.width = width,
+		.height = height,
+		.format = format,
+		.modifier = DRM_FORMAT_MOD_INVALID,
+		.n_planes = 1,
+		.offset[0] = offset0,
+		.stride[0] = stride0,
+		.fd[0] = fd,
+	};
+
+	struct wlr_drm_buffer *buffer = calloc(1, sizeof(*buffer));
+	if (buffer == NULL) {
+		close(fd);
+		wl_resource_post_no_memory(resource);
+		return;
+	}
+	wlr_buffer_init(&buffer->base, &buffer_impl, width, height);
+
+	buffer->resource = wl_resource_create(client, &wl_buffer_interface, 1, id);
```

Comment:
Should'nt `WLR_DRM_VERSION` be used instead of `1` to create the resource?

Reply:
No, `wl_buffer` is an exception here: since it's created by many different parent interfaces (`wl_shm`, linux-dmabuf, etc), it's stuck at version 1 and cannot be extended.

https://gitlab.freedesktop.org/wayland/wayland/-/blob/92038fa39497f0a481b38e623bd604feb6a01882/protocol/wayland.xml#L431

But that's a good reflex :)